### PR TITLE
Fix errors in StandardAnchorTags tool

### PR DESCRIPTION
**Pull request by BillWagner**
I noticed that the standard anchor tags tool didn't open a PR when we merged some PRs in our meeting yesterday, when it should have.

There were a couple bugs in the StandardAnchorTags tool. The main one is that I had edited the README.MD file by hand, and changed the text the tool searches for to replace the TOC.  I fixed that by adding an HTML comment in the README so it's more resilient for edits.

After that fix, this uncovered a second issue where one header had repeated spaces. Fix that as well.

Finally, fix an error exposed by the new action added in #249 (Thanks @Youssef1313). I'll add comments on that PR momentarily.

Once this is merged, the tool should re-run and update the section numbers in the draft-v6 branch best on the PRs we merged yesterday in our meeting. [This run](https://github.com/dotnet/csharpstandard/actions/runs/727373035) failed yesterday when we merged the PR that added `nameof` expressions. We should have had a new PR that updated the TOC and affected sections.

### CODE REVIEW

File: tools/StandardAnchorTags/TocSectionNumberBuilder.cs [hunk 1]
```diff
@@ -166,6 +172,11 @@ private SectionLink BuildSectionLink(SectionHeader header, string filename)
             header.level = level;
 
             // A few cases for section number:
+
+            if (string.IsNullOrWhiteSpace(fields[1]))
+            {
+                throw null;
```
Comment (jskeet) [inline tools/StandardAnchorTags/TocSectionNumberBuilder.cs]: Why *are* we throwing null here?

Comment (BillWagner) [reply tools/StandardAnchorTags/TocSectionNumberBuilder.cs]: Test code I meant to remove. (It helped me force the debugger to break on the right spot). Removing.

Comment (jskeet) [reply tools/StandardAnchorTags/TocSectionNumberBuilder.cs]: Phew :)

**Comment by jskeet** [comment]
Currently the comment for this PR ends with "Once this is merged, the tool should" - what should it do? Enquiring minds are eager to know!
(I haven't really got a sense of the PR yet.)

**Comment by BillWagner** [comment]
Sorry about that @jskeet I was interrupted while creating the PR and didn't get the description updated. That should be fixed now.

**Review by jskeet — APPROVED** [review]
Okay - sounds good now. Thanks!

**Pull request merged**